core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -164,13 +164,13 @@ private[server] class MetadataCache(brokerId: Int) extends Logging {
       aliveNodes.clear()
       aliveBrokers.clear()
       updateMetadataRequest.liveBrokers.asScala.foreach { broker =>
-        val nodes = new EnumMap[SecurityProtocol, Node](classOf[SecurityProtocol])
-        val endPoints = new EnumMap[SecurityProtocol, EndPoint](classOf[SecurityProtocol])
-        broker.endPoints.asScala.foreach { case (protocol, ep) =>
-          endPoints.put(protocol, EndPoint(ep.host, ep.port, protocol))
-          nodes.put(protocol, new Node(broker.id, ep.host, ep.port))
+        val nodes = new java.util.HashMap[ListenerName, Node]
```

Comment:
Ah, yes, I remember. Might be worth a comment so no one is tempted to change it.